lib/gettext-setup/gettext_setup.rb

```diff
@@ -38,6 +41,24 @@ def self.initialize(locales_path, options = {})
     Locale.set_default(default_locale)
   end
 
+  def self.build_translation_repository(project_name, options)
+    FastGettext::TranslationRepository.build(project_name,
+                                             :path => locales_path,
+                                             :type => options[:file_format] || :po,
+                                             :ignore_fuzzy => false)
+  end
+
+  def self.add_repository_to_chain(repository)
+    @@translation_repositories.push(repository)
+  end
+
+  def self.chain_contains?(repository)
+    @@translation_repositories.each do |in_chain|
+      return true if repository.instance_variable_get(:@name) == in_chain.instance_variable_get(:@name)
```

Comment:
the chain type is expecting an array so a hash won't work as the final form. something like this might work but it feels pretty wonky:
```ruby
  def self.add_repository_to_chain(project_name,options)
    unless @@translation_repositories.include? project_name
      repository = FastGettext::TranslationRepository.build(project_name,
                                               :path => locales_path,
                                               :type => options[:file_format] || :po,
                                               :ignore_fuzzy => false)
      @@translation_repositories.push(project_name)
      @@translation_repository_chain.push(repository)
    end
  end
```